lte/gateway/c/session_manager/test/test_local_enforcer.cpp

```diff
@@ -1453,29 +1453,110 @@ TEST_F(LocalEnforcerTest, test_rar_create_dedicated_bearer) {
-  insert_static_rule_with_qos(0, "m1", "rule1", 1);
-  insert_static_rule(0, "m1", "rule2");
+// Create multiple rules with QoS and assert dedicated bearers are created
+// Simulate a policy->bearer mapping from MME, both success + failure.
+// Simulate additional rule updates to trigger dedicated bearer deletions
+TEST_F(LocalEnforcerTest, test_dedicated_bearer_lifecycle) {
+  const std::string imsi       = "IMSI1";
+  const std::string session_id = "1234";
+
+  // Three rules with QoS & one without
+  insert_static_rule_with_qos(0, "m1", "rule1", 1);  // QCI=1
+  insert_static_rule_with_qos(0, "m1", "rule2", 2);  // QCI=2
+  insert_static_rule_with_qos(0, "m1", "rule3", 3);  // QCI=3
+  insert_static_rule(0, "m1", "rule4");
 
-  // test_cfg_ is initialized with QoSInfo field
-  test_cfg_.common_context.mutable_sid()->set_id("IMSI1");
+  // test_cfg_ is initialized with QoSInfo field w/ QCI 5
+  test_cfg_.common_context.mutable_sid()->set_id(imsi);
+  test_cfg_.common_context.set_apn("apn1");
+  auto lte_context = test_cfg_.rat_specific_context.mutable_lte_context();
+  lte_context->mutable_qos_info()->set_qos_class_id(5);
+  lte_context->set_bearer_id(32);  // linked_bearer_id
 
   CreateSessionResponse response;
   response.mutable_static_rules()->Add()->set_rule_id("rule1");
   response.mutable_static_rules()->Add()->set_rule_id("rule2");
+  response.mutable_static_rules()->Add()->set_rule_id("rule3");
+  response.mutable_static_rules()->Add()->set_rule_id("rule4");
 
   // expect only 1 rule in the request since only rules with a QoS field
   // should be mapped to a bearer
   EXPECT_CALL(
-      *spgw_client, create_dedicated_bearer(CheckCreateBearerReq("IMSI1", 1)))
+      *spgw_client, create_dedicated_bearer(CheckCreateBearerReq(imsi, 3)))
       .Times(1)
       .WillOnce(testing::Return(true));
 
   local_enforcer->init_session_credit(
-      session_map, "IMSI1", "1234", test_cfg_, response);
+      session_map, imsi, session_id, test_cfg_, response);
+
+  // Test successful creation of dedicated bearer for rule1 + rule2
+  auto bearer_bind_req_success1 =
+      create_policy_bearer_bind_req(imsi, 32, "rule1", 16);
+  auto bearer_bind_req_success2 =
+      create_policy_bearer_bind_req(imsi, 32, "rule2", 24);
+  std::unordered_set<std::string> rule_ids({"rule1", "rule2"});
+  // Expect NO call to PipelineD for rule1
+  EXPECT_CALL(
+      *pipelined_client,
+      deactivate_flows_for_rules(
+          imsi, CheckSubset(rule_ids), CheckCount(0), testing::_))
+      .Times(0);
+  auto update = SessionStore::get_default_session_update(session_map);
+  local_enforcer->bind_policy_to_bearer(
+      session_map, bearer_bind_req_success1, update);
+  local_enforcer->bind_policy_to_bearer(
+      session_map, bearer_bind_req_success2, update);
+
+  // Test unsuccessful creation of dedicated bearer for rule3 (bearer_id = 0)
+  auto bearer_bind_req_fail =
+      create_policy_bearer_bind_req(imsi, 32, "rule3", 0);
```

Comment:
can we define what 32 as a variable? Also bearer IDs for a UE are in the range of 5-15 with maximum of 11 bearers for UE (regardless of the number of sessions).